safe_transaction_service/history/tasks.py

```diff
@@ -472,3 +479,58 @@ def send_webhook_task(address: Optional[str], payload: Dict[str, Any]) -> int:
 
         sent_requests += 1
     return sent_requests
+
+
+@app.shared_task(
+    bind=True,
+    soft_time_limit=SOFT_TIMEOUT,
+    autoretry_for=(MetadataRetrievalExceptionTimeout,),
+    time_limit=LOCK_TIMEOUT,
+    retry_backoff=3,
+    retry_kwargs={"max_retries": 4},
+)
+def retry_get_metadata_task(self, address: str, id: int) -> bool:
+    """
+    Give more chances to get the metadata of collectible
+    """
+    from .services import CollectiblesServiceProvider
+
+    collectibles_service = CollectiblesServiceProvider()
+    redis_key = collectibles_service.get_redis_metadata_key(address, id)
+    redis = get_redis()
+    redis_collectible = redis.get(redis_key)
+
+    if not redis_collectible:
```

Comment:
Shouldn't you check that `redis_collectible` holds no metadata? Just in case this task is triggered at the same time from 2 different web workers and metadata is already retrieved by one worker

Reply:
Agree with the comments, but not sure if celery workers can take the same task at the same task, do you think that this is possible?